source/server/connection_handler_impl.cc

```diff
@@ -324,21 +324,32 @@ void ConnectionHandlerImpl::ActiveTcpListener::onAcceptWorker(
 
 void ConnectionHandlerImpl::ActiveTcpListener::newConnection(
     Network::ConnectionSocketPtr&& socket) {
+  auto stream_info = std::make_unique<StreamInfo::StreamInfoImpl>(parent_.dispatcher_.timeSource());
+  stream_info->setDownstreamLocalAddress(socket->localAddress());
+  stream_info->setDownstreamRemoteAddress(socket->remoteAddress());
+
   // Find matching filter chain.
   const auto filter_chain = config_.filterChainManager().findFilterChain(*socket);
   if (filter_chain == nullptr) {
     ENVOY_LOG(debug, "closing connection: no matching filter chain found");
     stats_.no_filter_chain_match_.inc();
+    stream_info->setResponseFlag(StreamInfo::ResponseFlag::NoRouteFound);
+    stream_info->onRequestComplete();
+    for (const auto& access_log : config_.accessLogs()) {
+      access_log->log(nullptr, nullptr, nullptr, *stream_info);
+    }
```

Comment:
Please consolidate this duplication into a helper function.

Reply:
Done